packages/fscomponents/src/components/serializable/SerializableImage.tsx

```diff
@@ -21,19 +22,28 @@ export interface SerializableImageProps
   > {
   href?: string;
   style?: ImageStyle;
-  onPress?: () => void;
+  onPress?: (href?: string) => void;
 }
 
 export const FSSerializableImage = React.memo<SerializableImageProps>(
-  ({ onPress, style, ...props }) => {
+  ({ onPress, style, href, ...props }) => {
     const [host, self] = extractHostStyles(style);
+    const navigator = useNavigator();
 
-    if (!onPress) {
+    const handlePress = (href?: string) => () => {
+      if (onPress) {
+        onPress(href);
+      } else if (href) {
+        navigator.open(href);
+      }
+    };
+
+    if (!(href || onPress)) {
       return <Image {...props} style={[self, host]} />;
     }
 
     return (
-      <TouchableOpacity style={host} onPress={onPress}>
+      <TouchableOpacity style={host} onPress={handlePress(href)}>
```

Comment:
We should just be able to use `href` directly from props rather then using a HOF here which should improve performance a bit as TouchableOpacity will no longer have it's onPress changing every render.